src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/VirtualMachineScaleSetExtensionImpl.cs

```diff
@@ -0,0 +1,210 @@
+/**
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT License. See License.txt in the project root for
+* license information.
+*/ 
+
+namespace Microsoft.Azure.Management.V2.Compute
+{
+
+    using Microsoft.Azure.Management.V2.Compute.VirtualMachineScaleSetExtension.Update;
+    using Microsoft.Azure.Management.V2.Resource.Core.ChildResource.Update;
+    using Microsoft.Azure.Management.V2.Resource.Core.ChildResource.Definition;
+    using Microsoft.Azure.Management.V2.Compute.VirtualMachineScaleSetExtension.Definition;
+    using Microsoft.Azure.Management.V2.Compute.VirtualMachineScaleSet.Definition;
+    using Microsoft.Azure.Management.V2.Compute.VirtualMachineScaleSet.Update;
+    using Microsoft.Azure.Management.V2.Resource.Core;
+    using System.Collections.Generic;
+    using Management.Compute.Models;
+    using Resource.Core.ChildResourceActions;
+
+    /// <summary>
+    /// Implementation of {@link VirtualMachineScaleSetExtension}.
+    /// </summary>
+    public partial class VirtualMachineScaleSetExtensionImpl  :
+        ChildResource<VirtualMachineScaleSetExtensionInner, VirtualMachineScaleSetImpl>,
+        IVirtualMachineScaleSetExtension,
+        IDefinition<IWithCreate>,
+        IUpdateDefinition<IWithApplicable>,
+        VirtualMachineScaleSetExtension.Update.IUpdate
+    {
+        private IDictionary<string,object> publicSettings;
+        private IDictionary<string,object> protectedSettings;
+        internal  VirtualMachineScaleSetExtensionImpl (VirtualMachineScaleSetExtensionInner inner, VirtualMachineScaleSetImpl parent) : base(inner.Id, inner, parent)
+        {
+            InitializeSettings();
+        }
+
+        public string Name
+        {
+            get
+            {
+                return this.Inner.Name;
+            }
+        }
+        public string PublisherName
+        {
+            get
+            {
+                return this.Inner.Publisher;
+            }
+        }
+        public string TypeName
+        {
+            get
+            {
+                return this.Inner.Type;
+            }
+        }
+        public string VersionName
+        {
+            get
+            {
+                return this.Inner.TypeHandlerVersion;
+            }
+        }
+        public bool? AutoUpgradeMinorVersionEnabled
+        {
+            get
+            {
+                return this.Inner.AutoUpgradeMinorVersion;
+            }
+        }
+        public IDictionary<string,object> PublicSettings
+        {
+            get
+            {
+                return this.publicSettings;
+            }
+        }
+        public string PublicSettingsAsJsonString
```

Comment:
better to either remove it if don't want to implement this, also put blank lines between properties. Install CodeMaid plugin on visualstudio it is awesome.